### PR TITLE
#33 - Support for Ruby Gems

### PR DESCRIPTION
**Pull request by bwmcadams**
### 💻 Description of Change(s) (w/ context)
*What did you change, and what additional context do reviewers need?****

Support for Ruby Gems, with tests
- The code right now for processing Gems is a duplicate of the code for Apache Commons archives; it is left as a separate method to allow us to do any extra handling/processing needed for the metadata and checksums inside a gem

### 🧠 Rationale Behind Change(s)
*Why were these changes made? What tradeoffs were considered?*

We need support for Ruby Gems in Goatrodeo

### 📝 Test Plan
*Beyond unit tests, how did you make sure this code is ready for production?*

Testing against more Gems in process to expand support validation

### 📜 Documentation
*What documentation did you add or update? 
Was the documentation appropriate for the scope of this change?*

Added inline documentation

### 💣 Quality Control
(All items must be checked before a PR is merged)
Did you…
- [x] Mention an issue number in the PR title?
- [x] Update the version # in the build file?
- [x] Create new and/or update relevant existing tests?
- [x] Create or update relevant documentation and/or diagrams?
- [x] Comment your code?
- [x] Fix any stray verbose logging (removing, or moving to debug / trace level)?

### Before Merging…
- Make sure the Quality Control boxes are all ticked
- Make sure any open comments or conversations on the PR are resolved

### CODE REVIEW

**Comment by bwmcadams** [comment]
Note this is branched off `wip_iso`, so until that PR is merged it will show all the changes from `wip_iso` as well as Gems

File: build.sbt [hunk 1]
```diff
@@ -22,12 +25,15 @@ lazy val root = project
     libraryDependencies += "commons-io" % "commons-io" % "2.16.1",
     libraryDependencies += "io.bullet" %% "borer-core" % "1.14.1",
     libraryDependencies += "io.bullet" %% "borer-derivation" % "1.14.1",
+    libraryDependencies += "com.palantir.isofilereader" % "isofilereader" % "0.6.1",
```
Comment (earldouglas) [inline build.sbt:28]: Is this PR stacked on #31, or should we separate it from the partial ISO support?

File: src/test/scala/GemFileSuite.scala [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+limitations under the License. */
+
+import com.typesafe.scalalogging.Logger
+import io.spicelabs.goatrodeo.util.{FileWalker, FileWrapper, GitOIDUtils}
+
+import java.io.File
+
+// For more information on writing tests, see
+// https://scalameta.org/munit/docs/getting-started.html
+class GemFileSuite extends munit.FunSuite {
```
Comment (dpp) [inline src/test/scala/GemFileSuite.scala:22]: I like breaking the tests into different files!

**Review by dpp — APPROVED** [review]
Looks good.

Please either update based on comments as well as fixing the merge conflicts or alternatively fix the merge conflicts, merge, and open tickets for the additional work (figuring out the PackageIdentifier metadata thing)

File: build.sbt (hunk 1 above)
Comment (bwmcadams) [reply build.sbt:28]: Stacked on #31, i forked from there…

**Review by dpp — APPROVED** [review]
Looks good.

Do we need to segregate Ruby GEMs as a different type identified by filename suffix or is there a way we can (1) identify it as a TAR (post opening with Apache Commons) and then (2) look for markers inside the TAR?

This is a meta-question meant to get stuff swirling in Brendan's brain rather than "this is the way it should be done"

**Review by earldouglas — APPROVED** [review]

**Review by earldouglas — APPROVED** [review]

**Pull request merged**